Notes/Algorithms/Data Structures/Hash-Tables.c

```diff
@@ -4,34 +4,231 @@
 
 #define SUCCESS 0
 #define ERROR -1
-#define OVERFLOW -2
-#define EMPTY -3
+
+#define TEST 9
+#define EXPECTED 10
+
+#define SIZE 10
 
 // Hash struct
-typedef struct Hash {
+typedef struct Item {
     int key;
     int value;
-} Hash;
+} Item;
 
-void print() {
+//create global testHash and expectedHash for testFunctions to use
+Item* testHash[SIZE];
+Item* expectedHash[SIZE];
 
+// traverse function is for visual confirmation of differences between hashTables when building functionality. 
+// void traverse() {
+//     for (int i = 0; i < SIZE; i++) {
+//         int value = testHash[i]->value;
+//         int key = testHash[i]->key;        
+//         printf("Test Key: %d Value: %d \n", key, value);
+//     }
+//     for (int i = 0; i < SIZE; i++) {
+//         int value = expectedHash[i]->value;
+//         int key = expectedHash[i]->key;
+//         printf("Expected Key: %d Value: %d \n", key, value);
+//     }
+// }
+
+//compare checks the testHash and expectedHash for parity
+// if the two HashTables match returns true
+// if a mismatch occurs returns false
+bool compare() {
+    for (int i = 0; i < SIZE; i++) {
+        int testKey = testHash[i]->key;
+        int testValue = testHash[i]->value;
+        int expectedKey = expectedHash[i]->key;
+        int expectedValue = expectedHash[i]->value;
+        if (testKey != expectedKey || testValue != expectedValue) {
+            return false;
+        }
+    }
+    return true;
 }
 
-void traverse() {
+// hashing takes in a key
+// returns a "hashed" index based on the SIZE of the hashTable
```

Comment:
Just some general advice when writing comments. Try to avoid including implementation details. If the `hashing` function implementation changes so that it no longer uses the `SIZE` value then the comment will need to be changed as well. Simply stating that this function returns a hashed index to be used for the hashTable would be better.